libraries/chain/db_block.cpp

```diff
@@ -746,5 +784,41 @@ bool database::before_last_checkpoint()const
 {
    return (_checkpoints.size() > 0) && (_checkpoints.rbegin()->first >= head_block_num());
 }
+   
+vector< custom_authority_object > database::get_custom_authorities_by_account( account_id_type account ) const
+{
+   const auto& authority_by_account = get_index_type<custom_authority_index>().indices().get<by_account>();
+   
+   vector<custom_authority_object> result;
+   
+   auto itr = authority_by_account.find(account);
+   while(itr != authority_by_account.end() && (*itr).account == account)
+   {
+      result.emplace_back(*itr++);
+   }
+   
+   return result;
+}
+   
+void database::verify_custom_authorities( const transaction& trx )const
+{
+   for (auto& op: trx.operations)
+   {
+      auto required_accounts = get_required_accounts(op);
+      for (auto& account_id: required_accounts)
+      {
+         auto custom_authorities = get_custom_authorities_by_account(account_id);
+         custom_authorities = filter_enabled_custom_authorities(custom_authorities);
+         
+         bool operation_validated = custom_authorities.empty();
```

Comment:
IMHO `_validated` is the wrong term. It should rather be `_authorized` or `_verified_`.

Reply:
done

Reply:
fixed that